File: backend/app/schemas/audit.py
"""Pydantic schemas for audit events."""

from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict

AuditAction = Literal[
    "scan.create",
    "scan.delete",
    "scheduled.create",
    "scheduled.update",
    "scheduled.delete",
    "domain.create",
    "domain.delete",
    "api_key.create",
    "api_key.revoke",
    "webhook.create",
    "webhook.delete",
    "webhook.test",
    "totp.enable",
    "totp.disable",
]

AuditStatus = Literal["success", "failure"]


class AuditEventOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int
    action: str
    target_type: str | None
    target_id: int | None
    status: str
    ip: str | None
    user_agent: str | None
    created_at: datetime


class AuditEventList(BaseModel):
    items: list[AuditEventOut]
    total: int
    page: int
    page_size: int

File: frontend/src/types/audit.ts
export type AuditAction =
  | "scan.create"
  | "scheduled.create"
  | "scheduled.update"
  | "scheduled.delete"
  | "domain.create"
  | "api_key.create"
  | "api_key.revoke"
  | "webhook.create"
  | "webhook.delete"
  | "webhook.test"
  | "totp.enable"
  | "totp.disable";

export type AuditStatus = "success" | "failure";

export interface AuditEvent {
  id: number;
  user_id: number;
  action: AuditAction;
  target_type: string | null;
  target_id: number | null;
  status: AuditStatus;
  ip: string | null;
  user_agent: string | null;
  created_at: string;
}

export interface AuditEventList {
  items: AuditEvent[];
  total: number;
  page: number;
  page_size: number;
}

export interface AuditFilters {
  page: number;
  pageSize: number;
  action?: AuditAction;
  status?: AuditStatus;
  dateFrom?: string; // YYYY-MM-DD
  dateTo?: string; // YYYY-MM-DD
}

export const ACTION_LABELS: Record<AuditAction, string> = {
  "scan.create": "Création scan",
  "scheduled.create": "Création scan planifié",
  "scheduled.update": "Mise à jour scan planifié",
  "scheduled.delete": "Suppression scan planifié",
  "domain.create": "Ajout domaine",
  "api_key.create": "Création clé API",
  "api_key.revoke": "Révocation clé API",
  "webhook.create": "Création webhook",
  "webhook.delete": "Suppression webhook",
  "webhook.test": "Test webhook",
  "totp.enable": "Activation 2FA",
  "totp.disable": "Désactivation 2FA",
};
